### PR TITLE
pkg: use `errors.Join` instead of appending to slice

### PR DESCRIPTION
**Pull request by shajmakh**
Instead of using a slice of error texts and converting it back to error
type or using a slice of errors, use the `errors` go library to `Join`
multiple errors into one.

### CODE REVIEW

**Comment by ffromani** [comment]
can `golang.org/x/sync/errgroup` be an alternative? just asking

**Comment by ffromani** [comment]
> golang.org/x/sync/errgroup

would not, is not related to the problem at hand (it's a sync subpkg)

**Comment by ffromani** [comment]
OTOH this is probably simpler than expected: https://golang.ch/migrate-from-hashicorps-go-multierror-to-standard-library-multierror-golang-1-20/

**Comment by shajmakh** [comment]
> @shajmakh: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> ci/prow/ci-e2e	[6c1e3b5](https://github.com/openshift-kni/numaresources-operator/commit/6c1e3b53e5710c1de5ed77751c95fa2f006f0403)	[link](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift-kni_numaresources-operator/1028/pull-ci-openshift-kni-numaresources-operator-main-ci-e2e/1849707068508868608)	true	`/test ci-e2e`
> ci/prow/ci-install-e2e	[6c1e3b5](https://github.com/openshift-kni/numaresources-operator/commit/6c1e3b53e5710c1de5ed77751c95fa2f006f0403)	[link](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift-kni_numaresources-operator/1028/pull-ci-openshift-kni-numaresources-operator-main-ci-install-e2e/1849707068617920512)	true	`/test ci-install-e2e`
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openshift-kni&repo=numaresources-operator&pr=1028). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:shajmakh).

an outage, will rerun after some while

**Review by ffromani — COMMENTED** [review]
/approve
/lgtm

**Review by ffromani — COMMENTED** [review]
do we need other updates besides `internal/dangling/dangling.go` ?

**Comment by ffromani** [comment]
> do we need other updates besides `internal/dangling/dangling.go` ?

probably `tools/verify-commit/...`

**Comment by shajmakh** [comment]
yes also` test/utils/padder`, thanks for the note. updating ..

File: controllers/numaresourcesoperator_controller.go [hunk 1]
```diff
@@ -449,17 +449,16 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 	klog.V(4).InfoS("RTESync start", "trees", len(trees))
 	defer klog.V(4).Info("RTESync stop")
 
-	errorList := dangling.DeleteUnusedDaemonSets(r.Client, ctx, instance, trees)
-	if len(errorList) > 0 {
-		klog.ErrorS(fmt.Errorf("failed to delete unused daemonsets"), "errors", errorList)
+	err := dangling.DeleteUnusedDaemonSets(r.Client, ctx, instance, trees)
+	if err != nil {
+		klog.ErrorS(fmt.Errorf("failed to delete unused daemonsets"), "errors", err)
```
Comment (ffromani) [inline controllers/numaresourcesoperator_controller.go:454]: this needs to be rewritten:
`klog.ErrorS(err, "failed to deleted unused daemonsets")`

can be done in a later PR

File: controllers/numaresourcesoperator_controller.go [hunk 2]
```diff
@@ -449,17 +449,16 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
...
 	}
 
-	errorList = dangling.DeleteUnusedMachineConfigs(r.Client, ctx, instance, trees)
-	if len(errorList) > 0 {
-		klog.ErrorS(fmt.Errorf("failed to delete unused machineconfigs"), "errors", errorList)
+	err = dangling.DeleteUnusedMachineConfigs(r.Client, ctx, instance, trees)
+	if err != nil {
+		klog.ErrorS(fmt.Errorf("failed to delete unused machineconfigs"), "errors", err)
```
Comment (ffromani) [inline controllers/numaresourcesoperator_controller.go:459]: same comment as above (`...unused machineconfigs`)

**Review by ffromani — COMMENTED** [review]
/approve

**Comment by ffromani** [comment]
another minor comment inside, PTAL

File: controllers/numaresourcesoperator_controller.go (hunk 1 above)
Comment (shajmakh) [reply controllers/numaresourcesoperator_controller.go]: updated. Thanks for your valuable review rounds

File: controllers/numaresourcesoperator_controller.go (hunk 2 above)
Comment (shajmakh) [reply controllers/numaresourcesoperator_controller.go]: updated. Thanks for your valuable review rounds

**Pull request merged**